### PR TITLE
Add more functionalities to donut chart

### PR DESCRIPTION
**Pull request by krkshitij**
## Related Issue(s)

- Fixes #

### CODE REVIEW

**Review by AtishayMsft — APPROVED** [review]

File: packages/web-components/src/donut-chart/donut-chart.ts [hunk 1]
```diff
@@ -123,15 +130,41 @@ export class DonutChart extends FASTElement {
...
+        tooltipBody.style['borderLeft'] = `4px solid ${d.data.color}`;
         legendText.textContent = d.data.legend;
         yText.textContent = d.data.data;
+        yText.style['color'] = d.data.color;
         tooltip.style['opacity'] = '1';
-        tooltip.style['left'] = `${event.pageX - rootDiv.getBoundingClientRect().left + window.scrollX}px`;
-        tooltip.style['top'] = `${event.pageY - rootDiv.getBoundingClientRect().top - window.scrollY}px`;
-        // tooltip.style['pointerEvents'] = 'none';
+
+        const rootBounds = rootDiv.getBoundingClientRect();
```
Comment (AtishayMsft) [inline packages/web-components/src/donut-chart/donut-chart.ts:161]: getBoundingClientRect could possibly be undefined

File: packages/web-components/src/donut-chart/donut-chart.styles.ts [hunk 2]
```diff
@@ -85,4 +91,41 @@ export const styles = css`
...
+    pointer-events: none;
+  }
+
+  .calloutBlockContainer {
+    padding-left: 8px;
+    color: ${colorNeutralForeground2};
+  }
+
+  .calloutLegendText {
+    font-family: ${fontFamilyBase};
```
Comment (AtishayMsft) [inline packages/web-components/src/donut-chart/donut-chart.styles.ts:118]: replace these group with typography defined in this file. https://github.com/microsoft/fluentui/blob/master/packages/web-components/src/styles/partials/typography.partials.ts

For eg all the font related styles here can be replaced with this.
https://github.com/microsoft/fluentui/blob/263ead68a05ac23499be137b41e02a5579522c08/packages/web-components/src/styles/partials/typography.partials.ts#L56

**Review by AtishayMsft — APPROVED** [review]

File: packages/web-components/src/donut-chart/donut-chart.ts [hunk 3]
```diff
@@ -123,15 +130,41 @@ export class DonutChart extends FASTElement {
...
-      path.setAttribute('aria-label', '.....lorem ipsum.....');
+      path.setAttribute('aria-label', `${d.data.legend}, ${d.data.data}.`);
       path.setAttribute('role', 'img');
       path.addEventListener('mouseover', event => {
+        if (this._selectedLegend !== '' && this._selectedLegend !== d.data.legend) {
+          return;
+        }
+
+        tooltipBody.style['borderLeft'] = `4px solid ${d.data.color}`;
+        legendText.textContent = d.data.legend;
```
[the remark this answers is not in the document]
Comment (krkshitij) [reply packages/web-components/src/donut-chart/donut-chart.ts:141]: yes, will add the checks

File: packages/web-components/src/donut-chart/donut-chart.styles.ts (hunk 2 above)
Comment (krkshitij) [reply packages/web-components/src/donut-chart/donut-chart.styles.ts]: done

File: packages/web-components/src/donut-chart/donut-chart.ts [hunk 4]
```diff
@@ -1,66 +1,308 @@
...
-    return 'donut-chart';
-  }
+  @attr({ converter: nullableNumberConverter })
+  public height: number = 200;
+
+  @attr({ converter: nullableNumberConverter })
+  public width: number = 200;
+
+  @attr({ attribute: 'hide-legends', mode: 'boolean' })
+  public hideLegends?: boolean;
```
Comment (Anush2303) [inline packages/web-components/src/donut-chart/donut-chart.ts:18]: Where are we using these attributes ?

Comment (krkshitij) [reply packages/web-components/src/donut-chart/donut-chart.ts:18]: Haven't used them yet

**Review by AtishayMsft — APPROVED** [review]

**Pull request merged**